module/module_interceptors_test.go

```diff
@@ -141,6 +144,52 @@ func TestOpID(t *testing.T) {
 	}
 }
 
+func TestModuleClientTimeoutInterceptor(t *testing.T) {
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	// Precompile module to avoid timeout issues when building takes too long.
+	modPath := rtestutils.BuildTempModule(t, "module/testmodule")
+	cfg := &config.Config{
+		Modules: []config.Module{{
+			Name:    "test",
+			ExePath: modPath,
+		}},
+		Components: []resource.Config{{
+			API:   generic.API,
+			Model: resource.NewModel("rdk", "test", "helper"),
+			Name:  "helper1",
+		}},
+	}
+	r, err := robotimpl.New(ctx, cfg, logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	helper1, err := r.ResourceByName(generic.Named("helper1"))
+	test.That(t, err, test.ShouldBeNil)
+
+	// Artificially lower default method timeout to an impossibly small duration.
+	origDefaultMethodTimeout := rdkgrpc.DefaultMethodTimeout
+	rdkgrpc.DefaultMethodTimeout = time.Nanosecond
```

Comment:
[minor] what you have is fine, I don't think it will ever flake, but we can make this 100% impossible by making the duration negative.

```suggestion
	// Artificially set default method timeout to have timed out in the past.
	origDefaultMethodTimeout := rdkgrpc.DefaultMethodTimeout
	rdkgrpc.DefaultMethodTimeout = -time.Nanosecond
```